modules/auxiliary/cloud/aws/enum_ec2.rb

```diff
@@ -56,28 +59,83 @@ def enumerate_regions
-    i.security_groups.each do |s|
+  def describe_ec2_instance(inst)
+    print_good "  #{inst.id} (#{inst.state.name})"
+    print_good "    Creation Date:  #{inst.launch_time}"
+    print_good "    Public IP:      #{inst.public_ip_address} (#{inst.public_dns_name})"
+    print_good "    Private IP:     #{inst.private_ip_address} (#{inst.private_dns_name})"
+    # Report hosts and info
+    mac_addr = inst.network_interfaces.select do |iface|
+      iface.private_ip_address == inst.private_ip_address
+    end.first.mac_address
+    iname = inst.tags.find { |t| t.key == 'Name' } ? inst.tags.find { |t| t.key == 'Name' }.value : inst.private_dns_name
+    iinfo = inst.tags.find { |t| t.key == 'Description' } ? inst.tags.find { |t| t.key == 'Description' }.value : nil
+    report_host(
+      host: inst.private_ip_address,
+      mac: mac_addr,
+      os_name: inst.platform_details,
+      os_flavor: inst.architecture,
+      name: iname,
+      info: iinfo,
+      comments: "ec2-id: #{inst.id} (#{inst.placement.availability_zone})"
+    )
+    if inst.public_ip_address
+      report_note(
+        host: inst.private_ip_address,
+        type: 'ec2.public_ip',
+        data: inst.public_ip_address
+      )
+    end
+    #eips = inst.network_interfaces.map {|i| i.association && i.association.public_ip}.compact # <-- works in pry, breaks at runtime in AWS SDK
+    #report_note(
+    #  host: inst.private_ip_address,
+    #  type: 'ec2.public_ips',
+    #  data: eips.join(' ')
+    #) unless eips.empty?
+    if inst.public_ip_address && !inst.public_dns_name.empty?
+      report_note(
+        host: inst.private_ip_address,
+        type: 'ec2.public_dns',
+        data: "#{inst.public_dns_name} #{inst.public_ip_address}"
+      )
+    end
+    if inst.hypervisor
+      report_note(
+        host: inst.private_ip_address,
+        type: 'ec2.hypervisor',
+        data: inst.hypervisor
+      )
+    end
+    inst.security_groups.each do |s|
       print_good "    Security Group: #{s.group_id}"
+      report_note(
+        host: inst.private_ip_address,
+        type: "ec2.#{s.group_id}",
+        data: s.group_name
+      )
+    end
+    inst.tags.each do |t|
+      print_good "    Tag: #{t.key} = #{t.value}"
+      report_note(
+        host: inst.private_ip_address,
+        type: "ec2.tag #{t.key}",
+        data: t.value
+      )
     end
   end
 
   def run
-    regions = datastore['REGION'] ? [datastore['REGION']] : regions = enumerate_regions()
+    regions = enumerate_regions
 
-    regions.each do |region|
+    regions.uniq.each do |region|
       vprint_status "Checking #{region}..."
       ec2 = Aws::EC2::Resource.new(
         region: region,
         access_key_id: datastore['ACCESS_KEY_ID'],
         secret_access_key: datastore['SECRET_ACCESS_KEY']
       )
 
-      instances = ec2.instances.limit(datastore['LIMIT'])
+      instances = datastore['LIMIT'] ? ec2.instances : ec2.instances.limit(datastore['LIMIT'])
```

Comment:
This is incorrect. When `datastore['LIMIT']` is truthy, you *do* want to use it.

```suggestion
      instances = datastore['LIMIT'] ? ec2.instances.limit(datastore['LIMIT']) : ec2.instances
```